### ISSUE TITLE
error parsing one line loop blocks if not last in the STAR file

### ISSUE THREAD

**Issue by inter1965** [alisterburt/starfile#2]
for a simple star file:
```
# version 30001

data_optics

loop_ 
_rlnVoltage #1 
_rlnSphericalAberration #2 
_rlnAmplitudeContrast #3 
_rlnOpticsGroup #4 
_rlnImageSize #5 
_rlnImagePixelSize #6 
_rlnImageDimensionality #7 
  300.000000     2.700000     0.070000            1          400     1.04000            2 
 

# version 30001

data_particles

loop_ 
_rlnImageName #1 
_rlnMicrographName #2 
_rlnCoordinateX #3 
_rlnCoordinateY #4 
_rlnAngleRot #5 
_rlnAngleTilt #6 
_rlnAnglePsi #7 
_rlnOriginXAngst #8 
_rlnOriginYAngst #9 
_rlnDefocusU #10 
_rlnDefocusV #11 
_rlnDefocusAngle #12 
_rlnPhaseShift #13 
_rlnCtfBfactor #14 
_rlnOpticsGroup #15 
_rlnRandomSubset #16 
_rlnClassNumber #17 
_rlnGroupNumber #18 
_rlnNormCorrection #19 
_rlnLogLikeliContribution #20 
_rlnMaxValueProbDistribution #21 
_rlnNrOfSignificantSamples #22 
000001@0001_sum_particles.mrcs 0001_sum.mrc   587.000000  1268.000000    -51.52270    99.622780    -21.58926    -24.39938    15.273376 12588.289062 12267.391602    36.674278     0.000000     0.000000            1            1            5            1     0.473071 69141.181865     0.993221            2 
```
try parsing with starfile, the result is as below:
```
In [1]: import starfile

In [2]: df = starfile.open("Stars/sample.star")

In [3]: df[0]
Out[3]:
Empty DataFrame
Columns: [rlnVoltage, rlnSphericalAberration, rlnAmplitudeContrast, rlnOpticsGroup, rlnImageSize, rlnImagePixelSize, rlnImageDimensionality]
Index: []
```
that means the parsing of the optics block is not successful.

**Comment by alisterburt** [comment]
There must be something wrong with the handling of 1 line loop blocks - sorry about that, I'll take a look!

**Comment by alisterburt** [comment]
okay, this seems to be the python engine of `pandas.read_csv()`, required when using the `skipfooter` argument, not handling `header=None` appropriately in this case. I've opened an issue at  pandas-dev/pandas##36827 and will try to cobble together a workaround when I've worked out exactly which situations cause this to fail

**Comment by inter1965** [comment]
Great!
Thanks a lot.

**Comment by alisterburt** [comment]
Fixed! Updated version (0.2.0) is available from PyPi

**Comment by alisterburt** [comment]
Found a better solution which avoids using the `skipfooter` argument of `pd.read_csv()`, allowing use of the C engine for all data blocks which should also increase speed. Update included in v0.2.1
